poc/ts/src/GetClass.ts

```
@@ -0,0 +1,61 @@
+  registerClass(identifier: string, classConstructor: Function): void {
+    this.classRegistry[identifier] = classConstructor;
+  }
+
+  /**
+   * Get what this object is becoming
+   *
+   * @param current Current object in metamorphosis chain
+   * @returns Next class(es) or null if transformation is complete
+   */
+  invoke(current: object): Function | Function[] | null {
+    const constructor = current.constructor;
```

Biome (2.1.2)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)